### PR TITLE
Update lsra-detail.md

### PR DESCRIPTION
**Pull request by CarolEidt**

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL

File: docs/design/coreclr/jit/lsra-detail.md [hunk 1]
```diff
@@ -308,6 +309,7 @@ After LSRA, the graph has the following properties:
         must issue the appropriate move.
 
     -   However, if such a node is constrained to a set of registers,
+        as in the case of x86 instructions which require a byte-addressible register,
```
Comment (kunalspathak) [inline docs/design/coreclr/jit/lsra-detail.md:312]: nit: addressable

File: docs/design/coreclr/jit/lsra-detail.md [hunk 2]
```diff
@@ -321,7 +323,12 @@ After LSRA, the graph has the following properties:
 
     -   Note that a write-thru variable def is always written to the stack, and the `GTF_SPILLED`
```
Comment (kunalspathak) [inline docs/design/coreclr/jit/lsra-detail.md:324]: while you are here: "and it is marked as `GTF_SPILLED`"

File: docs/design/coreclr/jit/lsra-detail.md [hunk 3]
```diff
@@ -321,7 +323,12 @@ After LSRA, the graph has the following properties:
...
         flag (not otherwise used for pure defs) to indicate that it also remains live
-        in the assigned register.
+        in the assigned register. This is somewhat counter-intuitive, but
```
Comment (kunalspathak) [inline docs/design/coreclr/jit/lsra-detail.md:326]: May be we should move this after we describe what a node with `GTF_SPILLED`  means and further re-phrase as :

"If a node has both `GTF_SPILL` and `GTF_SPILLED`, the tree node is spilled after it is evaluated (`GTF_SPILL`) and reloaded prior to the use (`GTF_SPILLED`). After the node is spilled and before the reload, its value is *not* live in the register. However, for EH-write-thru variables, there is an exception. The variable def is *always* written to the stack (`GTF_SPILL`), should always be reloaded (`GTF_SPILLED`) and *always* remain live in the register as well."

File: docs/design/coreclr/jit/lsra-detail.md [hunk 4]
```diff
@@ -487,6 +494,52 @@ node, which builds `RefPositions` according to the liveness model described abov
...
     -   For a `STORE_LCL_VAR` of a write-thru `lclVar`, the `RefPosition` is marked `writeThru`.
 
+    -   There is some special handling for `GT_PUTARG_REG` nodes whose source is a non-last-use lclVar.
+        At build time, we mark such intervals as `isSpecialPutArg`.
+        At allocation time, if the lclVar is already in the argument register, but it has another use
+        prior to the call, we don't want to reassign that register to the tree temp interval, since that
+        would require us to spill the lclVar and reload it for the next use.
+        Instead, we retain the assignment of the register to the lclVar, and mark that `RegRecord` as
+        `isBusyUntilNextKill` so that it isn't reused if the lclVar goes dead before the call.
+        (Otherwise, if the either the lclVar is in a different register, or if its next use is after
```
Comment (kunalspathak) [inline docs/design/coreclr/jit/lsra-detail.md:504]: nit: Otherwise, if the lclVar is...

File: docs/design/coreclr/jit/lsra-detail.md [hunk 5]
```diff
@@ -630,7 +688,30 @@ LinearScanAllocation(List<RefPosition> refPositions)
...
 -   Resolution
 
-    -   Perform resolution at block boundaries, adding moves as needed.
+    Since register allocation is performed over a linearized list of nodes, there
+    is no guarantee that register assignments on off-path edges will be consistent.
+    The resolution phase is responsible for performing moves, as needed to match up
+    the register assignments across edges.
+
+    -    The critical edges are handled first. These are the most problematic, as there
+         is no single location at which the moves can be added. We handle the critical
```
Comment (kunalspathak) [inline docs/design/coreclr/jit/lsra-detail.md:697]: This sentence doesn't sound right. Did you mean "We handle all the critical edges that originates from the source block"?

**Review by kunalspathak — APPROVED** [review]
LGTM with some small suggestions.

File: docs/design/coreclr/jit/lsra-detail.md (hunk 3 above)
Comment (CarolEidt) [reply docs/design/coreclr/jit/lsra-detail.md:326]: Sounds good, though the description should be reordered to make it clear that the reloading comes first in the general case.

File: docs/design/coreclr/jit/lsra-detail.md (hunk 5 above)
Comment (CarolEidt) [reply docs/design/coreclr/jit/lsra-detail.md:697]: No, all the critical edges are handled first. I'll work on a rephrasing of this.

**Comment by kunalspathak** [comment]
Changes looks good. Thanks!

**Pull request merged**